File: Difficulty: Easy/Identical Trees/identical-trees.cpp
/* A binary tree node


struct Node
{
    int data;
    struct Node* left;
    struct Node* right;

    Node(int x){
        data = x;
        left = right = NULL;
    }
};
*/

class Solution {
  public:
    // Function to check if two trees are identical.
    bool isIdentical(Node *r1, Node *r2) {
        // Your Code here
         if (r1 == NULL && r2 == NULL) {
            return true;
        }
        // If one of the nodes is NULL and the other is not, they are not identical
        if (r1 == NULL || r2 == NULL) {
            return false;
        }
        // Check if current node data is equal and recursively check left and right subtrees
        return (r1->data == r2->data) 
            && isIdentical(r1->left, r2->left) 
            && isIdentical(r1->right, r2->right);
    }
};

File: Difficulty: Easy/Identical Trees/identical-trees.py
'''
class Node:
    def _init_(self, val):
        self.right = None
        self.data = val
        self.left = None
'''
class Solution:
    def isIdentical(self,root1, root2):
        # code here
        if root1 is None and root2 is None:
            return True
        
        # Base case: one tree is empty, the other is not
        if root1 is None or root2 is None:
            return False
        
        # Check if current nodes have same data and recursively check left and right subtrees
        return (root1.data == root2.data) and \
               self.isIdentical(root1.left, root2.left) and \
               self.isIdentical(root1.right, root2.right)